client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -75,15 +80,110 @@ class EditorMediaModalDetailItem extends Component {
 
 		return (
 			<Button
-				className={ classNames( 'editor-media-modal-detail__edit', className ) }
+				className="editor-media-modal-detail__edit"
 				onClick={ onEdit }
-				disabled={ isItemBeingUploaded( item ) }
+				disabled={ isItemBeingUploaded( item ) || item.original_loaded }
 			>
 				<Gridicon icon="pencil" size={ 36 } /> { translate( 'Edit Image' ) }
 			</Button>
 		);
 	}
 
+	renderRestoreButton() {
+		const {
+			item,
+			onRestore,
+			onRestoreApply,
+			translate
+		} = this.props;
+
+		const isLoading = item.loading_original;
+		const isReadyToRestore = item.original_loaded;
+
+		const handleRestoringMethod = event => {
+			if ( isReadyToRestore ) {
+				return onRestoreApply( item, event );
+			}
+
+			return onRestore( item, event );
+		};
+
+		const buttonText = isReadyToRestore ? translate( 'Apply' ) : translate( 'Restore Original' );
+
+		return (
+			<Button
+				className={ classNames(
+					'editor-media-modal-detail__restore',
+					{ 'is-loading': isLoading },
+					{ 'is-ready': isReadyToRestore },
+				) }
+				disabled={ isLoading }
+				onClick={ handleRestoringMethod }
+			>
+				<Gridicon
+					icon={ isReadyToRestore ? 'checkmark' : 'refresh' }
+					size={ 36 } />
+					&nbsp;{ buttonText }
```

Comment:
Ok, just FYI we are doing this now:

![image](https://cloud.githubusercontent.com/assets/77539/19787163/ad3e1954-9c77-11e6-8c7f-d3ef69beace5.png)
https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/media-modal/detail/detail-item.jsx#L82